script/state_manager.js

```diff
@@ -0,0 +1,440 @@
+/*
+ * Module for handling States
+ *
+ * All states should be get and set through the StateManager ($SM).
+ *
+ * The manager is intended to handle all needed checks and error catching.
+ * This includes creating the parents of layered/deep states so undefined states
+ * do not need to be tested for and created beforehand.
+ *
+ * When a state is changed, an update event is sent out containing the name of the state
+ * changed or in the case of multiple changes (.setM, .addM) the parent class changed.
+ * Event: type: 'stateUpdate', stateName: <path of state or parent state>
+ *
+ * Original file created by: Michael Galusha
+ */
+
+var StateManager = {
+
+	MAX_STORE: 99999999999999,
+
+	options: {},
+
+	init: function(options) {
+		this.options = $.extend(
+				this.options,
+				options
+		);
+
+		//create categories
+		var cats = [
```

Comment:
The naming of the variable for the array could be changed to so that it is less confusing for the user as to what it's being used for.  Naming the variable categories is not much longer than cats and would explicitly let anyone who's looking at the code know what it's being used for. Also in a game that mimics a real life scenario, some might interpret the variable name literally and think it holds different types of cats.